src/Symfony/Component/Process/Process.php

```diff
@@ -998,9 +994,15 @@ private function getDescriptors()
 
         if (!$this->useFileHandles && $this->enhanceSigchildCompatibility && $this->isSigchildEnabled()) {
             // last exit code is output on the fourth pipe and caught to work around --enable-sigchild
-            $descriptors = array_merge($descriptors, array(array('pipe', 'w')));
+            $descriptors[3] = array('pipe', 'w');
+
+            $trap = '';
+            foreach (self::$posixSignals as $s) {
+                $trap .= "trap 'echo s$s >&3' $s;";
+            }
 
-            $this->commandline = '('.$this->commandline.') 3>/dev/null; code=$?; echo $code >&3; exit $code';
+            $this->commandline = $trap.'{ ('.$this->commandline.') <&3 3<&- 3>/dev/null & } 3<&0;';
```

Comment:
A comment explaining what's done in this commandline, with a link to the stackoverflow issue would be more than welcome.
I doubt anybody could read this without error in two years (thinking about the first implementation was years ago)